src/main/java/uk/gov/di/ipv/lambda/CredentialIssuerHandler.java

```diff
@@ -4,34 +4,63 @@
 import com.amazonaws.services.lambda.runtime.RequestHandler;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
 import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.oauth2.sdk.token.AccessToken;
 import com.nimbusds.oauth2.sdk.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.di.ipv.domain.CredentialIssuerException;
 import uk.gov.di.ipv.domain.ErrorResponse;
+import uk.gov.di.ipv.dto.CredentialIssuerConfig;
 import uk.gov.di.ipv.dto.CredentialIssuerRequestDto;
 import uk.gov.di.ipv.helpers.ApiGatewayResponseGenerator;
 import uk.gov.di.ipv.helpers.RequestHelper;
+import uk.gov.di.ipv.service.CredentialIssuerService;
 
+import java.net.URI;
 import java.util.Collections;
-import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
 public class CredentialIssuerHandler implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
 
-    private Set<String> validCredentialIssuers = Set.of("PassportIssuer", "FraudIssuer");
+    private static final Logger LOGGER = LoggerFactory.getLogger(CredentialIssuerHandler.class);
+
+    private final CredentialIssuerService credentialIssuerService;
+
+    private Set<CredentialIssuerConfig> credentialIssuers;
+
+    public CredentialIssuerHandler(CredentialIssuerService credentialIssuerService, Set<CredentialIssuerConfig> credentialIssuerConfig) {
+        this.credentialIssuerService = credentialIssuerService;
+        this.credentialIssuers = credentialIssuerConfig;
+    }
+
+    public CredentialIssuerHandler() {
+        CredentialIssuerConfig passportIssuer = new CredentialIssuerConfig("PassportIssuer", URI.create("http://www.example.com"));
+        CredentialIssuerConfig fraudIssuer = new CredentialIssuerConfig("FraudIssuer", URI.create("http://www.example.com"));
+        this.credentialIssuers = Set.of(passportIssuer, fraudIssuer);
+        this.credentialIssuerService = new CredentialIssuerService();
+    }
 
     @Override
     public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent input, Context context) {
 
-        Map<String, String> body = RequestHelper.parseRequestBody(input.getBody());
-        ObjectMapper objectMapper = new ObjectMapper();
-        CredentialIssuerRequestDto request = objectMapper.convertValue(body, CredentialIssuerRequestDto.class);
+        CredentialIssuerRequestDto request = RequestHelper.convertRequestBody(input.getBody(), CredentialIssuerRequestDto.class);
 
         var errorResponse = validate(request);
         if (errorResponse.isPresent()) {
             return errorResponse.get();
-        } else {
+        }
+
+        CredentialIssuerConfig credentialIssuerConfig = getCredentialIssuerConfig(request).get();
```

Comment:
Potential for a NoSuchElementException here. Its been recognised by sonar as a bug and we should probably be doing a getCredentialIssuerConfig(request).isPresent() or getCredentialIssuerConfig(request).isEmpty() check before calling the .get()